src/lib/market-service/zerion/zerion.ts

```diff
@@ -0,0 +1,53 @@
+import { type AssetId, fromAssetId } from '@shapeshiftoss/caip'
+import type {
+  FindAllMarketArgs,
+  HistoryData,
+  MarketCapResult,
+  MarketData,
+  MarketDataArgs,
+  PriceHistoryArgs,
+} from '@shapeshiftoss/types'
+import Axios from 'axios'
+import { setupCache } from 'axios-cache-interceptor'
+
+import type { MarketService } from '../api'
+import { DEFAULT_CACHE_TTL_MS } from '../config'
+import type { ZerionFungibles, ZerionMarketData } from './types'
+import { zerionMarketDataToMarketData } from './utils'
+
+const axios = setupCache(Axios.create(), { ttl: DEFAULT_CACHE_TTL_MS, cacheTakeover: false })
+
+export class ZerionMarketService implements MarketService {
+  baseUrl = 'https://zerion.shapeshift.com'
+
+  getZerionTokenMarketData = async (assetId: AssetId): Promise<ZerionMarketData | undefined> => {
+    const { assetReference } = fromAssetId(assetId)
+    const url = `${this.baseUrl}/fungibles/${assetReference}`
+    const { data: res } = await axios.get<ZerionFungibles>(url)
+    return res.data.attributes.market_data
+  }
+
+  async findAll(_args?: FindAllMarketArgs): Promise<MarketCapResult> {
+    // TODO - unimplemented
+    return await Promise.resolve({})
+  }
+
+  async findPriceHistoryByAssetId({
+    assetId: _assetId,
+    timeframe: _timeframe,
+  }: PriceHistoryArgs): Promise<HistoryData[]> {
+    // TODO - unimplemented
```

Comment:
For a follow-up PR, less important than getting custom asset market data ready before custom token marketing.